elide-async/src/test/java/com/yahoo/elide/async/operation/JsonAPITableExportOperationTest.java

```diff
@@ -0,0 +1,142 @@
+import com.yahoo.elide.async.service.storageengine.ResultStorageEngine;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.core.audit.Slf4jLogger;
+import com.yahoo.elide.core.datastore.DataStoreTransaction;
+import com.yahoo.elide.core.datastore.inmemory.HashMapDataStore;
+import com.yahoo.elide.core.dictionary.EntityDictionary;
+import com.yahoo.elide.core.security.User;
+import com.yahoo.elide.core.security.checks.Check;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class JsonAPITableExportOperationTest {
+
+    private HashMapDataStore dataStore;
+    private User user;
+    private RequestScope requestScope;
+    private Elide elide;
+    private AsyncExecutorService asyncExecutorService;
+    private ResultStorageEngine engine;
+
+    @BeforeEach
+    public void setupMocks(@TempDir Path tempDir) {
+        dataStore = new HashMapDataStore(TableExport.class.getPackage());
+        Map<String, Class<? extends Check>> map = new HashMap<>();
+        map.put(AsyncAPIInlineChecks.AsyncAPIOwner.PRINCIPAL_IS_OWNER,
+                AsyncAPIInlineChecks.AsyncAPIOwner.class);
+        map.put(AsyncAPIInlineChecks.AsyncAPIAdmin.PRINCIPAL_IS_ADMIN,
+                AsyncAPIInlineChecks.AsyncAPIAdmin.class);
+        map.put(AsyncAPIInlineChecks.AsyncAPIStatusValue.VALUE_IS_CANCELLED,
+                AsyncAPIInlineChecks.AsyncAPIStatusValue.class);
+        map.put(AsyncAPIInlineChecks.AsyncAPIStatusQueuedValue.VALUE_IS_QUEUED,
+                AsyncAPIInlineChecks.AsyncAPIStatusQueuedValue.class);
+
+        elide = new Elide(
+                    new ElideSettingsBuilder(dataStore)
+                        .withEntityDictionary(new EntityDictionary(map))
+                        .withAuditLogger(new Slf4jLogger())
+                        .withDownloadApiPath("/export")
+                        .build());
+        user = mock(User.class);
+        requestScope = mock(RequestScope.class);
+        asyncExecutorService = mock(AsyncExecutorService.class);
+        engine = new FileResultStorageEngine(tempDir.toString());
+        when(asyncExecutorService.getElide()).thenReturn(elide);
+        when(asyncExecutorService.getResultStorageEngine()).thenReturn(engine);
+        when(requestScope.getApiVersion()).thenReturn(NO_VERSION);
+        when(requestScope.getUser()).thenReturn(user);
+        when(requestScope.getElideSettings()).thenReturn(elide.getElideSettings());
+        when(requestScope.getBaseUrlEndPoint()).thenReturn("https://elide.io");
+    }
+
+    @Test
+    public void testProcessQuery() throws URISyntaxException, IOException  {
+        dataPrep();
```

Comment:
I can't see the difference between this test setup and the next one.  What makes one succeed where the other fails?

Reply:
there is a difference in Query. /tableExport vs /tableExport1